### DISCUSSION TITLE
Can add multiple private accounts on the Github Android APP？

### DISCUSSION THREAD

**Discussion by dingdangdog** [community/community#38185] [Mobile] [answered]
I found that only one personal account can be configured on the APP. If I want to switch to another account, I can only log out first and then log in.

Can you consider adding multiple accounts in the Settings? If you want to switch accounts, you only need to select from the Settings.

**Comment by AJ0545282** [comment]
I second this, as I have two accounts (professional and personal)

**Comment by ghost** [comment]
Yes i also think this would be a nice option for the same reason as @AJ0545282 described.

**Comment by nrosu34** [comment]
Yes, please add the capacity to allow multiple accounts in the app so it can properly be used for security updates.

**Comment by jsoref** [comment]
Technically multiple free accounts for the same human are probably a violation of GitHub's terms of use. 

You're allowed to have 🤖 accounts, and a single account for your use. I know that Microsoft has in recent memory encouraged their own employees to discontinue the practice of having distinct work/personal accounts. 

> One person or legal entity may maintain no more than one free Account (if you choose to control a machine account as well, that's fine, but it can only be used for running a machine).
https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#b-account-terms

Now, that doesn't mean that GitHub's Mobile application has to make this experience unfriendly, but, if this is what you're doing and if I were a PM for the project, I wouldn't prioritize this...

(Dead accounts/accounts for which you've lost your credentials are fine as the keyword in the terms is `maintain`. You aren't maintaining dead accounts.)

---

Out of curiosity, what use case are you addressing that requires you to animate multiple accounts?

**Reply by dingdangdog** [reply]
**Thanks for your reply, you said is very reasonable.**

I have multiple emails, So I want to divide my study and work into different accounts. If this idea is not allowed by Microsoft, and I Sorry. I don't know that.

**Reply by jsoref** [reply]
You can link multiple email addresses to a single account and can configure different repositories/organizations to send notifications to different email addresses.

You could convert one of your accounts into an organization and add your other account into the organization.

**Reply by dingdangdog** [reply]
Sounds good. I'll try. Thanks~

**Reply by churbain** [reply]
My company ask for a specific id as name (annoying), and we're not actually allowed to contribute to repository outside the organisation.

So what do I do  ?

**Reply by jsoref** [reply]
As long as they pay for a pro account for it, that appears to be acceptable (the issue is multiple active free accounts). But, will you really need to use both accounts here? If they're that concerned, I'd half expect them to want you to use a dedicated computer in which case this shouldn't be relevant.

**Reply by churbain** [reply]
Interesting point. 

I haven't tried connecting with the account on the new (more secure) organization)

**Reply by unaiur** [reply]
The company I work for also requires an account with a specific name. It is more related to reputation than security.

**Reply by jsoref** [reply]
Fwiw, Having worked for 3 companies that have issued me cell phones (and paid for corporate cell phone service for them), I'd half expect these companies to issue a cell phone to you (just as, presumably, they issue you a company laptop for _company business only_).

**Comment by joel-zutobi** [comment]
+1

**Comment by dzmpr** [comment]
@GitHub can you provide feedback on this request? There are several posts about this feature. It's long-awaited feature for me.

**Reply by candyho** [reply]
Hey, multi-account support on mobile is something we are actively working on. Please stay tuned for more updates in the future. Thanks for bearing with us!

**Comment by Syedzain125** [comment]
nice app

**Comment by nhstanley** [comment]
The Gmail app supports the ability to use multiple work and personal accounts on the same app, and also use for MFA, without issues. Surprising GitHub can't seem to handle this properly.

**Comment by DiegoBM** [comment]
I reckon that you can also use GitHub as a Dual or Parallel App in Android (each vendor has a different name for the feature). It's not the best solution but it works if you need to use two accounts

**Comment by aisgbnok** [comment, marked as answer]
[It looks like they have finally added support today!](https://github.blog/changelog/2023-08-15-log-in-with-multiple-github-accounts-on-github-mobile/) [Update your app.](https://play.google.com/store/apps/details?id=com.github.android)

**Reply by paramsiddharth** [reply]
Masha-Allah! <3 Finally. Thank you, GitHub!

**Reply by churbain** [reply]
It's just me or it's iOS only ??

**Reply by dingdangdog** [reply]
wohoo, nice!

**Reply by DiegoBM** [reply]
I can't see it on Android either. No new updates in the store, maybe they 're rolling that out in batches for Android?

**Reply by PITR-DEV** [reply]
I'm on the beta branch and I did receive the android update already. The feature is here and working

**Reply by churbain** [reply]
> I'm on the beta branch and I did receive the android update already. The feature is here and working 

Good to know. Guess I'll just have to be a little patient

**Reply by eliperkins** [reply]
Hey folks, the update is rolling out on the Play Store gradually this week. Keep your eyes peeled on the Play Store for an update coming your way soon!

**Comment by eliperkins** [comment]
Hey @dingdangdog and folks, we've shipped functionality this week to allow you to log in with multiple accounts, like your personal and work accounts, at a the same time! [Read more about it in our Changelog post](https://github.blog/changelog/2023-08-15-log-in-with-multiple-github-accounts-on-github-mobile/).

<img width="1200" alt="image" src="https://github.com/community/community/assets/1051453/d2ba902e-2199-4625-948a-532d64a6f44e">

Download or update GitHub Mobile today from the [Apple App Store](https://apps.apple.com/us/app/github/id1477376905) or [Google Play Store](https://play.google.com/store/apps/details?id=com.github.android) to get started.

**Reply by dingdangdog** [reply]
NICE!

**Comment by dingdangdog** [comment]
NICE!

> Hey @dingdangdog and folks, we've shipped functionality this week to allow you to log in with multiple accounts, like your personal and work accounts, at a the same time! [Read more about it in our Changelog post](https://github.blog/changelog/2023-08-15-log-in-with-multiple-github-accounts-on-github-mobile/).
> 
> <img alt="image" width="1200" src="https://private-user-images.githubusercontent.com/1051453/261077985-d2ba902e-2199-4625-948a-532d64a6f44e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNjEwNDYsIm5iZiI6MTcxNDM2MDc0NiwicGF0aCI6Ii8xMDUxNDUzLzI2MTA3Nzk4NS1kMmJhOTAyZS0yMTk5LTQ2MjUtOTQ4YS01MzJkNjRhNmY0NGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjlUMDMxOTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGZiYTBjYmZhMTIyMjZiOGQ0MDEyNjdjNzE0ZWJmMGQ5MDE4NjdkY2FjYjhkNTZiYWMyNmZmMmMzYWI5YmE1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.VOLggtI4CX2lR4T9sDeBEYOAdlQlkhxE4Y1i1aBim6g">
> Download or update GitHub Mobile today from the [Apple App Store](https://apps.apple.com/us/app/github/id1477376905) or [Google Play Store](https://play.google.com/store/apps/details?id=com.github.android) to get started.
